js/src/rive.ts

```diff
@@ -1454,8 +1454,12 @@ export class Rive {
     if (this.eventCleanup !== null) {
       this.eventCleanup();
     }
-    this.artboard.delete();
-    // TODO: delete animation and state machine instances
+    // Delete all animation and state machine instances
+    this.stop();
+    if (this.artboard) {
```

Comment:
Is this just for sanity or was there a case where we were calling cleanup multiple times too?

Reply:
sanity - in case multiple cleanups are called by consumers. I'm seeing some funny business around how React calls cleanup on useEffects, which is where we call `rive.cleanup()`. But also seems a safeguard change to add in case others do this too.